consensus/spos/bls/subroundBlock.go

```diff
@@ -94,17 +106,27 @@ func (sr *subroundBlock) doBlockJob(ctx context.Context) bool {
 		return false
 	}
 
-	sentWithSuccess := sr.sendBlock(header, body)
-	if !sentWithSuccess {
-		return false
-	}
-
 	leader, errGetLeader := sr.GetLeader()
 	if errGetLeader != nil {
 		log.Debug("doBlockJob.GetLeader", "error", errGetLeader)
 		return false
 	}
 
+	sentWithSuccess := sr.sendBlock(header, body, leader)
+	if !sentWithSuccess {
+		return false
+	}
+
+	if sr.EnableEpochsHandler().IsFlagEnabled(common.ConsensusPropagationChangesFlag) {
+		marshalledHeader, err := sr.Marshalizer().Marshal(header)
```

Comment:
we have a function called CalculateHash in mx-chain-core-go that does the marshaling & hashing in one step

Reply:
:100: updated